#rectangle number
int32 rect_size
moving_points[] moving_pointsArray
